sky/data/storage_utils.py

```diff
@@ -63,6 +66,41 @@ def format_storage_table(storages: List[Dict[str, Any]],
         return 'No existing storage.'
 
 
+def get_excluded_files_from_skyignore(src_dir_path: str) -> List[str]:
+    """List files and patterns ignored by the .skyignore file
+    in the given source directory.
+    """
+    excluded_list: List[str] = []
+    expand_src_dir_path = os.path.expanduser(src_dir_path)
+    skyignore_path = os.path.join(expand_src_dir_path,
+                                  constants.SKY_IGNORE_FILE)
+
+    try:
+        with open(skyignore_path, 'r', encoding='utf-8') as f:
+            for line in f:
+                line = line.strip()
+                if line and not line.startswith('#'):
+                    if '*' in line:
+                        # Make parsing consistent with rsync.
+                        if line.startswith('/*'):
+                            line = '.' + line
```

Comment:
To allow `/dir/*.txt`, should we use `startswith('/')` instead of `startswith('/*')`?
```suggestion
                        if line.startswith('/'):
                            line = '.' + line
```

Reply:
Actually user should do `dir/*.txt`, `/dir/*.txt` means it's the absolute path. Updated docs to reflect this

Reply:
Hmm, it is a bit confusing to me. Do you mean `/test.txt` is for the current folder, but `/dir` is the absolute path at root?

Reply:
Ignore my previous implementation. Now both /test.txt and /dir are for current folder